src/app/pokemon/components/pokemon-gallery/pokemon-gallery.component.ts

```diff
@@ -1,31 +1,143 @@
 
-  constructor(private PokemonDataService: PokemonDataService) { }
+  constructor(
+    private PokemonDataService: PokemonDataService,
+    private store: Store<State>,
+    public dialog: MatDialog,
+    private _snackBar: MatSnackBar
+  ) {
+    this.loadPokemonsFromApi();
+  }
 
   ngOnInit(): void {
-    this.getPokemonsData();
-    console.log('Lista de pokemones', this.pokemonsList)
-  }
-  getPokemonsData() {
-    this.sub = this.PokemonDataService.getPokemonsData().subscribe({
-      next: pokemonsList => {
-        this.pokemonsList = pokemonsList;
-      },
-      error: err => this.errorMessage = err
+    this.getPokemonsDataFromStore();
+    this.isSearchingSubscription();
+  }
+
+  pokemonDescriptionUrl = (url: string) => {
+    return `${environment.POKEMONDATAAPI}pokemon-species/${url.split('/')[6]}/`;
+  };
+
+  selectPokemon(pokemon: Pokemon) {
+    this.store.dispatch(PokemonActions.selectedPokemons({ pokemon }));
+    this.store.dispatch(PokemonActions.loadPokemonsDescription({ pokemon }));
+    this.store.dispatch(
+      PokemonActions.loadPokemonsDescriptionGender({ pokemon })
+    );
+    this.store.dispatch(PokemonActions.maximumNumberOfFavoritesUnSelected());
+    this.getSelectedPokemonsFromStore();
+    this.openDialog(this.selectedPokemons.length);
+  }
+  openDialog(amountSelectedPokemons: any) {
+    setTimeout(() => {
+      if (amountSelectedPokemons > 1) {
+        const dialogRef = this.dialog.open(PokemonsModalVsComponent, {
+        });
+        dialogRef.afterClosed().subscribe(() => this.getKeepSelectedFromStore());
+      } else {
+        this.maxFavoritesSelected = false;
+        const dialogRef = this.dialog.open(PokemonsModalComponent, {
+        });
+        dialogRef.afterClosed().subscribe(() => this.getKeepSelectedFromStore());
+      }
+    }, 350);
+    
+  }
+
+  getSelectedPokemonsFromStore() {
+    this.store.select(getSelectedPokemons).subscribe((selectedPokemons) => {
+      if (selectedPokemons) {
+        this.selectedPokemons = selectedPokemons;
+      }
     });
   }
 
+  isSearchingSubscription() {
+    this.store
+      .select(getIsSearching)
+      .pipe(
+        map((isSearching: boolean) => {
+          this.isSearching = isSearching;
+          this.getPokemonsDataFromStore();
+        })
+      )
+      .subscribe();
+  }
+
+  getPokemonsDataFromStore() {
+    this.store.select(getPokemonsInfo).subscribe((pokemons) => {
+      if (pokemons) {
+        this.rootPokemonList = pokemons.rootPokemonList;
+        this.queriedPokemons = pokemons.queriedPokemons;
+      }
+    });
+  }
+
+  loadPokemonsFromApi(): void {
+    this.store.dispatch(PokemonActions.loadPokemons());
+    this.getPokemonsDataFromStore();
+  }
+
+  openToast() {
+    const snackBarRed = this._snackBar.openFromComponent(ToastComponent);
+  }
+
+  getKeepSelectedFromStore() {
+    this.store
+      .select(getPokemonsInfo)
+      .pipe(
+        first(),
+        map((pokemon) => {
```

Comment:
map should return a new version of the object

Reply:
Map should return a new version of the object for the next step on the pipe, if you want to have side effects use tap, in this specific scenario it is better to use a subscribe instead